lib/buffer.js

```diff
@@ -140,6 +146,15 @@ const bufferWarning = 'Buffer() is deprecated due to security and usability ' +
                       'Buffer.allocUnsafe(), or Buffer.from() methods instead.';
 
 function showFlaggedDeprecation() {
+  if (hardened) {
+    if (bufferWarningAlreadyEmitted) return;
+    process.emitWarning(
+      bufferWarning + ' Buffer() will soon throw in hardened mode.',
+      'DeprecationWarning', 'DEP0XXX');
+    bufferWarningAlreadyEmitted = true;
```

Comment:
Should we just throw in hardened mode? Why emitting a deprecation?

Reply:
+1 to throwing instead.

Reply:
I had a concern that it could introduce unexpected throws without prior deprecation and/or would hinder the usage of hardened mode. The plan was to keep it like that until a major version that could turn that into a throw, so that there would be an additional grace period.

Will take another look at the data and think about it more.

Reply:
@mcollina @cjihrig It should not throw by default, but there is an option to disable that and turn into that into warnings.

Reply:
@mcollina @cjihrig It should now throw by default, but there is an option to disable that and turn into that into warnings.